### PR TITLE
connman: add version 1.33 (new recipe)

### PR DESCRIPTION
**Pull request by sknsean**
Signed-off-by: Sean Nyekjaer sean@nyekjaer.dk

### CODE REVIEW

**Comment by sknsean** [comment]
Depends on https://github.com/oe-lite/base/pull/400

File: recipes/connman/connman-1.33/0001-plugin.h-Change-visibility-to-default-for-debug-symb.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
+From 4ddaf78dad5a9ee4a0658235f71b75132192123e Mon Sep 17 00:00:00 2001
+From: Khem Raj <raj.khem@gmail.com>
+Date: Sat, 7 Apr 2012 18:52:12 -0700
+Subject: [PATCH] plugin.h: Change visibility to default for debug symbols
+
+gold refuses to link in undefined weak symbols which
```
Comment (kimrhh) [inline recipes/connman/connman-1.33/0001-plugin.h-Change-visibility-to-default-for-debug-symb.patch]: we don't need patches for a linker we don't use :)

File: recipes/connman/connman.inc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+inherit autotools-autoreconf pkgconfig
+
+SRC_URI = "https://www.kernel.org/pub/linux/network/connman/connman-${PV}.tar.xz \
+           file://0001-plugin.h-Change-visibility-to-default-for-debug-symb.patch \
+           file://connman \
+           file://no-version-scripts.patch \
+           file://includes.patch \
+           "
+
+SRC_URI:>HOST_LIBC_musl += "file://0002-resolve-musl-does-not-implement-res_ninit.patch"
```
Comment (kimrhh) [inline recipes/connman/connman.inc]: could you please move all the musl specific patches here

File: recipes/connman/connman.inc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+DEPENDS:>USE_connman_gnutls += " libgnutls"
+DEPENDS_${PN}:>USE_connman_gnutls += " libgmp libgnutls libhogweed libnettle libtasn1 libz"
+RDEPENDS_${PN}:>USE_connman_gnutls += " libgmp libgnutls libhogweed libnettle libtasn1 libz"
+
+RECIPE_FLAGS += "connman_wifi"
+DEFAULT_USE_connman_wifi = "1"
+EXTRA_OECONF += "${EXTRA_OECONF_WIFI}"
+EXTRA_OECONF_WIFI = " --disable-wifi"
+EXTRA_OECONF_WIFI:USE_connman_wifi = " --enable-wifi ac_cv_path_WPASUPPLICANT=${sbindir}/wpa_supplicant"
+DEPENDS:>USE_connman_wifi += " wpa-supplicant-wpa-supplicant"
```
Comment (kimrhh) [inline recipes/connman/connman.inc]: doesn't wpa_supplicant set up some proper provides you can use instead of depending directly on the package?

Comment (sknsean) [inline recipes/connman/connman.inc]: is wpa-supplicant-wpa-supplicant the same as util/wpa-supplicant?

Comment (kimrhh) [inline recipes/connman/connman.inc]: seems to be the only content in the wpa-supplicant-wpa-supplicant package, so yes

File: recipes/connman/connman.inc [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+DEFAULT_USE_connman_gnutls = "0"
+GNUTLS_SUPPORT = " --disable-wispr"
+GNUTLS_SUPPORT:USE_connman_gnutls = " --enable-wispr"
+DEPENDS:>USE_connman_gnutls += " libgnutls"
+
+RECIPE_FLAGS += "connman_wifi"
+DEFAULT_USE_connman_wifi = "1"
+WIFI_SUPPORT = " --disable-wifi"
+WIFI_SUPPORT:USE_connman_wifi = " --enable-wifi"
+DEPENDS:>USE_connman_wifi += " wpa-supplicant-wpa-supplicant"
```
Comment (sknsean) [inline recipes/connman/connman.inc]: added openvpn and ofono but we are still missing an recipe for it :+1:

File: recipes/connman/connman.inc (hunk 3 above)
Comment (sknsean) [inline recipes/connman/connman.inc]: done

**Pull request merged**